src/arrayMethodJoin.js

```diff
@@ -4,9 +4,22 @@
  * Implement method join
  */
 function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
-    // write code here
+  [].__proto__.join2 = function(separator = ',') {
+    let joinResult = '';
+    const joinSeparator = String(separator);
```

Comment:
it's redundant here, you shouldn't create separate variable for separator, argument variable will be enough